packages/jest-snapshot/src/utils.js

```diff
@@ -179,3 +189,111 @@ export const saveSnapshotFile = (
+};
+
+const saveSnapshotsForFile = (
+  snapshots: InlineSnapshot[],
+  sourceFilePath: Path,
+) => {
+  const sourceFile = fs.readFileSync(sourceFilePath, 'utf8');
+  const config = prettier.resolveConfig.sync(sourceFilePath);
+  const {inferredParser} = prettier.getFileInfo.sync(sourceFilePath);
+
+  const newSourceFile = prettier.format(
+    sourceFile,
+    Object.assign({}, config, {
+      filepath: sourceFilePath,
+      parser: createParser(snapshots, inferredParser),
+    }),
+  );
+
+  if (newSourceFile !== sourceFile) {
+    fs.writeFileSync(sourceFilePath, newSourceFile);
+  }
+};
+
+const groupSnapshotsBy = (createKey: InlineSnapshot => string) => (
+  snapshots: InlineSnapshot[],
+) =>
+  snapshots.reduce((object, inlineSnapshot) => {
+    const key = createKey(inlineSnapshot);
+    return Object.assign(object, {
+      [key]: (object[key] || []).concat(inlineSnapshot),
+    });
+  }, {});
+
+const groupSnapshotsByFrame = groupSnapshotsBy(
+  ({frame: {line, column}}) => `${line}:${column - 1}`,
+);
+const groupSnapshotsByFile = groupSnapshotsBy(({frame: {file}}) => file);
+
+const createParser = (snapshots: InlineSnapshot[], inferredParser: string) => (
+  text: string,
+  parsers: {[key: string]: (string) => any},
+  options: any,
+) => {
+  // Workaround for https://github.com/prettier/prettier/issues/3150
+  options.parser = inferredParser;
+
+  const groupedSnapshots = groupSnapshotsByFrame(snapshots);
+  const remainingSnapshots = new Set(snapshots.map(({snapshot}) => snapshot));
+  let ast = parsers[inferredParser](text);
+
+  // flow uses a 'Program' parent node, babel expects a 'File'.
+  if (ast.type !== 'File') {
+    ast = file(ast, ast.comments, ast.tokens);
+    delete ast.program.comments;
+  }
+
+  traverse(ast, {
+    CallExpression({node: {arguments: args, callee}}) {
+      if (
+        callee.type !== 'MemberExpression' ||
+        callee.property.type !== 'Identifier'
+      ) {
+        return;
+      }
+      const {line, column} = callee.property.loc.start;
+      const snapshotsForFrame = groupedSnapshots[`${line}:${column}`];
+      if (!snapshotsForFrame) {
+        return;
+      }
+      if (snapshotsForFrame.length > 1) {
+        throw new Error(
+          'Jest. Multiple inline snapshots for the same call are not supported.',
+        );
+      }
+      const snapshotIndex = args.findIndex(
+        ({type}) => type === 'TemplateLiteral',
+      );
+      const values = snapshotsForFrame.map(({snapshot}) => {
+        remainingSnapshots.delete(snapshot);
+
+        return templateLiteral(
+          [templateElement({raw: escapeBacktickString(snapshot)})],
+          [],
+        );
+      });
+      const replacementNode = values[0];
+
+      if (snapshotIndex > -1) {
+        args[snapshotIndex] = replacementNode;
+      } else {
+        args.push(replacementNode);
+      }
+    },
+  });
+
+  if (remainingSnapshots.size) {
+    throw new Error(`Jest. Couldn't locate all inline snapshots.`);
```

Comment:
`Jest: `

Reply:
When would this error happen? Do you think users would see this often? If yes, can we polish this error message?

Reply:
It should never happen. Maybe if there's a parser bug (locations in the AST are incorrect), or if something is catastrophically wrong with sourcemaps.